### PR TITLE
Implement Paint Building

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This is v. v. similar to paint industry. Could probably share more code.

### CODE REVIEW

File: src/OpenLoco/src/Paint/PaintBuilding.cpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+        bool isMultiTile = buildingObj->hasFlags(BuildingObjectFlags::largeTile);
+
+        // Note: Image offsets will change as you move up the building but bboffset/size does not
+        const World::Pos3 imageOffset = (isMultiTile ? kImageOffsetBase2x2 : kImageOffsetBase1x1) + World::Pos3{ 0, 0, baseHeight };
+        const World::Pos3 bbOffset = (isMultiTile ? kBBOffsetBase2x2 : kBBOffsetBase1x1) + World::Pos3{ 0, 0, baseHeight };
+        const World::Pos3 bbSize = (isMultiTile ? kBBSizeBase2x2 : kBBSizeBase1x1) + World::Pos3{ 0, 0, bbLengthZ };
+
+        session.resetLastPS(); // Odd...
+        if (buildingObj->hasFlags(BuildingObjectFlags::hasShadows))
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Paint/PaintBuilding.cpp:173]: Interestingly no object seems to have hasShadows. I just know what it does from the industry object.

**Review by AaronVanGeffen — APPROVED** [review]
I've not that much to say about the code, but buildings look fine in-game.

**Comment by duncanspumpkin** [comment]
Missing drawing the animated lifts!

**Review by AaronVanGeffen — APPROVED** [review]
Buildings are looking good to me. Elevators moving up and down as expected.

**Pull request merged**